src/features/device-logs/lib/backends/loki.ts

```diff
@@ -384,23 +407,36 @@ export class LokiBackend implements DeviceLogsBackend {
 		}
 	}
 
-	private fromDeviceLogsToEntries(
-		ctx: LogContext,
-		logs: Array<InternalDeviceLog & { version?: number }>,
-	) {
-		const structuredMetadata = this.getStructuredMetadata(ctx);
+	private fromDeviceLogsToEntries(ctx: LogContext, logs: InternalDeviceLog[]) {
+		const deviceId = new loki.LabelPairAdapter()
+			.setName('device_id')
+			.setValue(`${ctx.id}`);
 		return logs.map((log) => {
 			const timestamp = new loki.Timestamp();
 			timestamp.setSeconds(Math.floor(Number(log.nanoTimestamp / 1000000000n)));
 			timestamp.setNanos(Number(log.nanoTimestamp % 1000000000n));
-			// store log line as JSON
-			const logJson = JSON.stringify(
-				{ ...log, version: VERSION },
-				omitNanoTimestamp,
-			);
+			const structuredMetadata = [
+				deviceId,
+				new loki.LabelPairAdapter()
+					.setName('timestamp')
```

Comment:
This is also questionable as to whether we should store it as structured metadata because every entry for it will be unique and possibly make a bunch of other things more painful and is probably not ergonomic or even useful to query by